vision/imageloader/imageloader.go

```diff
@@ -43,30 +44,38 @@ type ImageLoader struct {
 	err        error
 	pinMemory  bool
 	colorSpace string
+	// shuffle buffer configuration
+	bufSize     int
+	shuffleChan chan sample
+	seed        int64
 }
 
 // New returns an ImageLoader
 func New(fn string, vocab map[string]int, trans *transforms.ComposeTransformer,
-	mbSize int, pinMemory bool, colorSpace string) (*ImageLoader, error) {
+	mbSize, bufSize int, seed int64, pinMemory bool, colorSpace string) (*ImageLoader, error) {
 	r, e := tgz.OpenFile(fn)
 	if e != nil {
 		return nil, e
 	}
 	trans1, trans2 := splitComposeByToTensor(trans)
 	m := &ImageLoader{
-		r:          r,
-		vocab:      vocab,
-		sampleChan: make(chan sample, mbSize*4),
-		mbChan:     make(chan miniBatch, 4),
-		errChan:    make(chan error, 1),
-		trans1:     trans1,
-		trans2:     trans2,
-		mbSize:     mbSize,
-		pinMemory:  pinMemory,
-		colorSpace: colorSpace,
+		r:           r,
+		vocab:       vocab,
+		sampleChan:  make(chan sample, mbSize*4),
+		mbChan:      make(chan miniBatch, 4),
+		shuffleChan: make(chan sample, mbSize*4),
```

Comment:
We should expose the shuffle buffer size argument, `4*mbSize` may be small for ImageNet training.

Reply:
L66 is the channel buffer size, and L65 is the shuffle buffer size .